src/openms/source/PROCESSING/CENTROIDING/PeakPickerIM.cpp

```diff
@@ -2,35 +2,183 @@
+    }
+
     void PeakPickerIM::pickIMTraces(MSSpectrum& spectrum)
-    {        
-        // determine IM format of spectrum
+    {
+        // Debugging: Print the input spectrum size
+        std::cout << "Processing spectrum with " << spectrum.size() << " peaks." << std::endl;
+
+        /*
+        // IM format determination (Temporarily commented out)
         IMFormat format = IMTypes::determineIMFormat(spectrum);
         switch (format)
         {
             case IMFormat::NONE:
                 return; // no IM data
             case IMFormat::CENTROIDED:
                 return; // already centroided
-            case IMFormat::CONCATENATED:
-                // TODO call peak picking algorithm for concatenated IM data
-
-                // set format to centroided
-                spectrum.setIMFormat(IMFormat::CENTROIDED);
-                break;
             case IMFormat::UNKNOWN:
-                throw Exception::InvalidValue(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "IMFormat set to UNKNOWN after deterineIMFormat. This should never happen. Please contact the developers.", String(NamesOfIMFormat[(size_t)format]));
-            default:
-                throw Exception::InvalidValue(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Unknown IMFormat", String(NamesOfIMFormat[(size_t)format]));
+                throw Exception::InvalidValue(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION,
+                    "IMFormat set to UNKNOWN after determineIMFormat. This should never happen.",
+                    String(NamesOfIMFormat[(size_t)format]));
+        }
+        */
+
+        // Initialize resampling
+        double sampling_rate = computeOptimalSamplingRate(spectrum) * 4;
+        std::cout << "Using sampling rate: " << sampling_rate << std::endl;
+
+        // Set up custom parameters for the resampler
+        bool ppm = false;
+        Param resampler_param;
+        resampler_param.setValue("spacing", sampling_rate, "Spacing of the resampled output peaks.");
+        resampler_param.setValue("ppm", ppm ? "true" : "false", "Whether spacing is in ppm or Th");
+
+        LinearResamplerAlign lin_resampler;
+        lin_resampler.setParameters(resampler_param);
+
+        lin_resampler.raster(spectrum);
+        std::cout << "Size of resampled spectrum: " << spectrum.size() << std::endl;
+
+        // Print resampled peaks for debugging
+        for (const auto& peak : spectrum)
+        {
+            std::cout << "m/z: " << peak.getMZ()
+                      << ", intensity: " << peak.getIntensity() << std::endl;
         }
+
+        // Apply PeakPickerHiRes
+        PeakPickerHiRes picker;
+        picker.setParameters(parameters_);
+        MSSpectrum picked_spectrum;
+
+        picker.pick(spectrum, picked_spectrum);
+        std::cout << "Size of picked spectrum: " << picked_spectrum.size() << std::endl;
+
+        // Replace original spectrum with processed version
+        spectrum = picked_spectrum;
```

Comment:
do you annotate the average IM value somewhere? IM information is lost at this point, right?